integration/test/alert/alert.go

```diff
@@ -1,55 +1,133 @@
+			alertsData[alertData] = alertsData[alertData] + 1
+			totalAlerts++
 		}
 	}
-	return alertNames, response
+	return alertsData, totalAlerts
 }
 
-func GetAlertRules(alertRuleNames *[]string, alertRules *[]string, dir string, fileName string) {
-	alertRulesFilePath := dir + "/" + fileName
-	log.Info().Str("alertRulesFilePath", alertRulesFilePath).Msg("alert rules file path")
+func GetEmsAlerts(dir string, fileName string) ([]EmsData, []EmsData) {
+	totalEms := make([]EmsData, 0)
+	bookendEms := make([]EmsData, 0)
+	bookendEmsCount := 0
+
+	emsConfigFilePath := dir + "/" + fileName
+	log.Debug().Str("emsConfigFilePath", emsConfigFilePath).Msg("")
 
-	data, err := tree.ImportYaml(alertRulesFilePath)
+	data, err := tree.ImportYaml(emsConfigFilePath)
 	if err != nil {
 		utils.PanicIfNotNil(err)
 	}
 
-	for _, v := range data.GetChildS("groups").GetChildren() {
-		if v.GetNameS() == "rules" {
-			for _, a := range v.GetChildren() {
-				if a.GetNameS() == "alert" {
-					alertname := a.GetContentS()
-					*alertRuleNames = append(*alertRuleNames, alertname)
-				}
-				if a.GetNameS() == "expr" {
-					alertexp := a.GetContentS()
-					*alertRules = append(*alertRules, alertexp)
-				}
+	for _, child := range data.GetChildS("events").GetChildren() {
+		emsName := child.GetChildContentS("name")
+		if resolveEms := child.GetChildS("resolve_when_ems"); resolveEms != nil {
+			bookendEms = append(bookendEms, EmsData{EmsName: emsName, resolvingEms: resolveEms.GetChildContentS("name")})
+			bookendEmsCount++
+		}
+		totalEms = append(totalEms, EmsData{EmsName: emsName})
+	}
+
+	log.Debug().Msgf("Total ems configured: %d, Bookend ems configured:%d", len(totalEms), bookendEmsCount)
+	return totalEms, bookendEms
+}
+
+func GenerateEvents(emsNames []EmsData) []string {
+	supportedEms := make([]string, 0)
+	addr, user, pass := GetPollerDetail()
+	url := "https://" + addr + "/api/private/cli/event/generate"
+	method := "POST"
+
+	volumeArwCount := 0
+	vserverArwCount := 0
+	for _, e := range emsNames {
+		value := "1"
+		ems := ""
+		if ems = e.resolvingEms; ems == "" {
+			ems = e.EmsName
+		}
+		if ems == "arw.volume.state" {
+			value = volumeArwState[volumeArwCount]
+			volumeArwCount++
+		}
+		if ems == "arw.vserver.state" {
+			value = vserverArwState[vserverArwCount]
+			vserverArwCount++
+		}
+
+		jsonValue := []byte(fmt.Sprintf(`{"message-name": "%s", "values": [%s,2,3,4,5,6,7,8,9]}`, ems, value))
+		var data map[string]interface{}
+		data = utils.SendPostReqAndGetRes(url, method, jsonValue, user, pass)
+		if response := data["error"]; response != nil {
+			errorDetail := response.(map[string]interface{})
+			code := errorDetail["code"].(string)
+			target := errorDetail["target"].(string)
+			if !(code == "2" && target == "message-name") {
+				supportedEms = append(supportedEms, e.EmsName)
 			}
+		} else {
+			supportedEms = append(supportedEms, e.EmsName)
 		}
 	}
+
+	return supportedEms
+}
+
+func GetPollerDetail() (string, string, string) {
```

Comment:
minor but seems like you could use
conf.LoadHarvestConfig
return conf.PollerNamed() instead of destructing the poller?

Reply:
done